internal/infraenv/infraenv_test.go

```diff
@@ -96,118 +118,132 @@ var _ = Describe("Delete inactive infraenvs", func() {
+		inactiveInfraEnv3 := registerInfraEnvAtTime(emptyClusterID, t1)
 
-		activeInfraEnv1 := registerInfraEnv("")
-		activeInfraEnv2 := registerInfraEnv("")
-		activeInfraEnv3 := registerInfraEnv("")
+		activeInfraEnv1 := registerInfraEnvAtTime(emptyClusterID, t3)
+		activeInfraEnv2 := registerInfraEnvAtTime(emptyClusterID, t3)
+		activeInfraEnv3 := registerInfraEnvAtTime(emptyClusterID, t3)
 
-		Expect(state.DeleteOrphanInfraEnvs(ctx, 10, lastActive)).ShouldNot(HaveOccurred())
+		Expect(state.DeleteOrphanInfraEnvs(ctx, 10, strfmt.DateTime(t2))).Should(Succeed())
 
-		Expect(wasDeleted(db, *inactiveInfraEnv1.ID)).To(BeTrue())
-		Expect(wasDeleted(db, *inactiveInfraEnv2.ID)).To(BeTrue())
-		Expect(wasDeleted(db, *inactiveInfraEnv3.ID)).To(BeTrue())
+		expectWasDeleted(inactiveInfraEnv1.ID)
+		expectWasDeleted(inactiveInfraEnv2.ID)
+		expectWasDeleted(inactiveInfraEnv3.ID)
 
-		Expect(wasDeleted(db, *activeInfraEnv1.ID)).To(BeFalse())
-		Expect(wasDeleted(db, *activeInfraEnv2.ID)).To(BeFalse())
-		Expect(wasDeleted(db, *activeInfraEnv3.ID)).To(BeFalse())
+		expectWasNotDeleted(activeInfraEnv1.ID)
+		expectWasNotDeleted(activeInfraEnv2.ID)
+		expectWasNotDeleted(activeInfraEnv3.ID)
 	})
 
 	It("Delete inactive infraEnvs with new infraEnvs - limited", func() {
-		inactiveInfraEnv1 := registerInfraEnv("")
-		inactiveInfraEnv2 := registerInfraEnv("")
-		inactiveInfraEnv3 := registerInfraEnv("")
-		inactiveInfraEnv4 := registerInfraEnv("")
-		inactiveInfraEnv5 := registerInfraEnv("")
-		inactiveInfraEnv6 := registerInfraEnv("")
-
-		// To verify that lastActive is greater than the updatedAt field of inactiveInfraEnv3
-		time.Sleep(time.Millisecond)
-		lastActive := strfmt.DateTime(time.Now())
-
-		Expect(state.DeleteOrphanInfraEnvs(ctx, 3, lastActive)).ShouldNot(HaveOccurred())
-
-		Expect(wasDeleted(db, *inactiveInfraEnv1.ID)).To(BeTrue())
-		Expect(wasDeleted(db, *inactiveInfraEnv2.ID)).To(BeTrue())
-
-		Expect(wasDeleted(db, *inactiveInfraEnv3.ID)).To(BeFalse())
-		Expect(wasDeleted(db, *inactiveInfraEnv4.ID)).To(BeFalse())
-		Expect(wasDeleted(db, *inactiveInfraEnv5.ID)).To(BeFalse())
-		Expect(wasDeleted(db, *inactiveInfraEnv6.ID)).To(BeFalse())
+		t1 := defaultTime
+		t2 := defaultTime.Add(time.Second * 1)
+		t3 := defaultTime.Add(time.Second * 2)
+
+		inactiveInfraEnv1 := registerInfraEnvAtTime(emptyClusterID, t1)
+		inactiveInfraEnv2 := registerInfraEnvAtTime(emptyClusterID, t1)
+		inactiveInfraEnv3 := registerInfraEnvAtTime(emptyClusterID, t1)
+		inactiveInfraEnv4 := registerInfraEnvAtTime(emptyClusterID, t1)
+		inactiveInfraEnv5 := registerInfraEnvAtTime(emptyClusterID, t3)
+		inactiveInfraEnv6 := registerInfraEnvAtTime(emptyClusterID, t3)
+
+		Expect(state.DeleteOrphanInfraEnvs(ctx, 3, strfmt.DateTime(t2))).Should(Succeed())
+
+		expectWasDeleted(inactiveInfraEnv1.ID)
+		expectWasDeleted(inactiveInfraEnv2.ID)
+		expectWasDeleted(inactiveInfraEnv3.ID)
+		expectWasNotDeleted(inactiveInfraEnv4.ID)
```

Comment:
>Why should we expect that this particular infraenv was the one that was not deleted?

Because maxDelete is 3, and there are 4 infraenvs that expired (given that t2 is the expiration date), so ...

ah your question just clicked.

Yeah, you're right, I get your point now